.github/workflows/deploy-release.yml

```diff
@@ -0,0 +1,97 @@
+name: Publish Next Release (Manual)
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: Reference git commit to release (e.g. "master", "12341fa")
+        required: true
+        default: master
+
+env:
+  NODE_VERSION: 18
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.ref }}
+
+      - name: AWS, credentials setup
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-west-1
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      - name: Add INPUT_SHA env var
+        run: |
+          export INPUT_SHA=$(git rev-parse ${{ github.event.inputs.ref }})
+          echo "INPUT_SHA=`echo $INPUT_SHA`" >> $GITHUB_ENV
+
+      - name: Install branch-github-actions
+        uses: actions/checkout@v4
+        with:
+          repository: BranchMetrics/branch-github-actions
+          ref: master
+          path: .branch-github-actions
+          token: ${{ secrets.BRANCHLET_ACCESS_TOKEN }}
+
+      - name: Get next release version
+        uses: actions/github-script@v7
+        id: next-version
+        with:
+          result-encoding: string
+          script: |
+            const getNextVersion = require('./.branch-github-actions/custom-scripts/next-version');
+            const nextVersion = await getNextVersion({
+              core,
+              github,
+              context,
+              sha: process.env.INPUT_SHA,
+            });
+            return nextVersion;
+        env:
+          INPUT_SHA: ${{ env.INPUT_SHA }}
+
+      - name: Install Node ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Write version to files
+        run: |
+          ./deployment/write-versions.sh ${{ steps.next-version.outputs.result }}
+
+      - name: Configure NPM
+        run: npm ci
+
+      - name: Release to npm, s3, prod
+        run: |
+          ./deployment/release.sh ${{ steps.next-version.outputs.result }}
+
+      # We deploy to stage to send the updated release numbers. The code should not be any different
+      - name: Release to stage
```

Comment:
wouldnt we want to do this first? before prod?

Reply:
sure

Reply:
i mean like as a separate step

Reply:
kind of like how we deploy applications to staging, then verify, then manually release to prod

Reply:
could we stage the gha release and new version, and then have a manual action to put this on the prod cdn? does that make sense for something like this?

Reply:
we do that already

Reply:
sorry, i was confused

on master merge we deploy to staging

im deploying to staging here now so that staging has the same "version". The code will not change, just the version number so that prod and stage look more synced.

I tried to capture this in the comment but ill make it more clear

Reply:
i didnt want to increment the version every time in staging, that would be another solutoin i suppose

Reply:
my worry would be something like: adds major commit -> updates major version. Finds bug, tags commit major again because we had to change some other part of the api. Now, when we deploy to prod we have incremented 2 versions when we prob should have just done 1

Reply:
For manual pre-production "publications" we use `alpha` and `beta` suffixes. Example being https://www.npmjs.com/package/react-native-branch/v/6.4.0-alpha.0

Reply:
ah gotcha, sorry, was unaware we were already deploying to staging separately. what youve got looks good, unless maybe we can use the beta/alpha tags like gabe is indicating?

Reply:
that is an interesting idea, but i think i came up with a better solution

I am just always going to calculate the next version in staging and deploy that. I.e:

current: 2.84, next 2.85

every staging deploy, we check, get 2.84, and deploy 2.85. The github release wont change tho!

on prod, we deploy 2.85, and write 2.85 to releaeses.

Then, next stage deploy we will deploy 2.86 over and over... and on and on